### DISCUSSION TITLE
Password reset on Admin UI

### DISCUSSION THREAD

**Discussion by SashaMihalache** [keystonejs/keystone#7672] [Feature requests]
Hey guys,

Is there a way to set up the Admin Sign-in Page to support the `Reset password` feature out of the box?
It would be helpful for admins to be able to reset their passwords in case they forget them.

I know that in the `createAuth` if we define the `passwordResetLink` property, Graphql will expose 2 useful mutations to send and redeem the password change request.

Any thoughts on this?

**Comment by dcousens** [comment]
@SashaMihalache what would be your expectations for that feature if implemented in `keystone-6`?

A `Reset password` flow often integrates multiple different systems, many of which may be external and or proprietary.
That isn't a foreign concept, for example the [`cloudinary` field](https://github.com/keystonejs/keystone/tree/main/packages/cloudinary); but it should be a seamless developer experience; so I'm interested in your feedback :eyes:

**Reply by SashaMihalache** [reply]
Hi @dcousens 

I'd imagine the flow something like this:

1. User clicks a 'forgot your password?' CTA on the admin UI auth page, then inputs his email on a different page and a reset password link & token will be sent via email (here it could be done the same way the above mentioned mutations work in createAuth and the mail can be sent with whichever mail transport the developer wants).
2. When the user clicks on the email link, he gets redirected to a simple change your password page  (password and password confirm should suffice) on the admin side, where the token is checked

**Reply by SashaMihalache** [reply]
So based on this, I think the minimum effort would require:
1. A way to add a `forgot password` CTA on the admin auth page. Currently, I'm not sure if there's a way to do that
![Group 136](https://user-images.githubusercontent.com/7697631/178196998-32c7f8e3-f330-42d2-a0b4-1d57f6eaf614.png)
2. An intermediary page prompts you to input your email to send you that reset link (behind-the-scenes it's the `sendUserPasswordResetLink` mutation being fired).
3. A page to come back to from the email link that uses the `redeemUserPasswordResetToken` to do validations and change the password on the `User` model and on success redirects you to the admin auth page

**Reply by SashaMihalache** [reply]
@dcousens any thoughts on this?

**Reply by dcousens** [reply]
Hi @SashaMihalache,  this feedback is awesome.   We haven't circled back around to this yet but we intend to :+1:

**Reply by joakimtrulsson** [reply]
Hi @dcousens, Any updates on this? Looking in the "keystone-6-auth-pages-SigninPage.esm.js" file and I can see that there is a "forgot password" mode.
